Documentation/GettingStarted.md

```diff
@@ -1,104 +1,861 @@
+1. [Error handling](#error-handling)
+1. [Debugging](#debugging)
+1. [UI layer tips](#ui-layer-tips)
+1. [Examples](Examples.md)
 
-To understand these examples, you will need to understand to `>-` operator.
+# Observables aka Sequences
 
-This is the definition of `>-` operator
+## Basics
+This equivalence is the most important thing to understand.
+
+Sequences are a simple concept.
+
+Everybody is familiar with sequences. Lists/sequences are probably one of the first concepts mathematicians/programmers learn.
+
+Here is a sequence of numbers
+
+
+```
+--1--2--3--4--5--6--| // it terminates normally
+```
+
+Here is another one with characters
+
+```
+--a--b--a--a--a---d---X // it terminates with error
+```
+
+Some sequences are finite, and some are infinite, like sequence of button taps
+
+```
+---tap-tap-------tap--->
+```
+
+These diagrams are called marble diagrams.
+
+[http://rxmarbles.com/](http://rxmarbles.com/)
+
+If everything is a sequence and every operation is just a transformation of input sequence into output sequence then it's pretty straightforward to compose operations.
+
+The concept of sequences lifts a lot of cognitive overload. You stop worrying about asynchronous events and how to compose them and think in terms of synchronous operations over sequences.
+
+People are creatures with huge visual cortexes. When you can visualize something easily, it's a lot easier to reason about it.
+
+If we were to specify sequence grammar as regular expression it would look something like this
+
+**Next* (Error | Completed)**
+
+This describes the following:
+
+* sequences can have 0 or more elements
+* once `Error` or `Completed` event is received sequence can't produce any other element
+
+Sequences in Rx are described by a push interface (aka callback).
 
 ```swift
-func >- <In, Out>(lhs: In, rhs: In -> Out) -> Out {
-    return rhs(lhs)
+enum Event<Element>  {
+    case Next(Element)      // next element of a sequence
+    case Error(ErrorType)   // sequence failed with error
+    case Completed          // sequence terminated successfully
+}
+
+class Observable<Element> {
+    func subscribe(observer: Observer<Element>) -> Disposable
+}
+
+protocol ObserverType {
+    func on(event: Event<Element>)
 }
 ```
-More practical explanation
+
+The semantics of `Event` values is pretty natural. There is one for next element, for error event and completed event.
+
+If you are curios [why `ErrorType` isn't generic](DesignRationale.md#why-error-type-isnt-generic).
+
+`ObserverType` is equivalent to `(Event<Element>) -> Void`
+
+So the only thing left on the table is `Disposable`.
+
+```
+protocol Disposable
+{
+    func dispose()
+}
+```
+
+## Disposing
+
+There is one additional way an observed sequence can terminate. When for some reason subscriber doesn't want to receive next elements of sequence and wants to release all of the resources that were allocated to compute those elements it just calls `dispose` method for his subscription.
```

Comment:
Or

> When for some reason subscriber doesn't want to receive next elements of sequence and wants to release all of the resources that were allocated to compute those elements it just calls `dispose` method for his subscription.

When you are done with a sequence and want to release all of the resources that were allocated to compute upcoming elements. Calling `dispose` on a subscription will clean this up for you.